cli/src/cmd/forge/script/mod.rs

```diff
@@ -699,6 +700,19 @@ impl ScriptConfig {
     fn target_contract(&self) -> &ArtifactId {
         self.target_contract.as_ref().expect("should exist after building")
     }
+
+    /// Checks if the RPCs used point to Ethereum Mainnet.
+    /// If not, warns the user that Shanghai is not supported.
+    async fn check_shanghai_support(&self) -> eyre::Result<()> {
+        for rpc in &self.total_rpcs {
+            let provider = ethers::providers::Provider::<Http>::try_from(rpc)?;
+            let chain_id = provider.get_chainid().await?;
+            if chain_id != U256::one() {
+                shell::println(format!("Shanghai is only supported on Ethereum Mainnet (Chain ID 1). This RPC uses chain id {} therefore contracts using PUSH0 will not work.", chain_id))?;
```

Comment:
I'd change this to a more meaningful description. The average user does not think in terms of assembly. I would instead reference the EIP-3855 and provide a [link](https://eips.ethereum.org/EIPS/eip-3855)

Reply:
Also shanghai is also supported on goerli and sepolia, not just mainnet

Reply:
Yeah agreed this message could be better—and probably a bit louder as well. I'm still not 100% sold on only erroring actually though; I've seen some [discussion around the topic](https://twitter.com/0xfoobar/status/1652673725542612993) and we might wanna consider instead erroring (with some mechanism to bypass this). we'll have the same story in cancun with transient storage so that's why I'm giving it some more thought

Reply:
did some improvements, here's an example of how it looks:

![Screen Shot 2023-05-02 at 13 49 03](https://user-images.githubusercontent.com/26014927/235744745-dafece09-689e-4aa7-b028-754f0a83a49c.png)

It maybe should be even bigger—lmk what you think about copy and if we should make it even louder!